instagram_search_by.py

```diff
@@ -0,0 +1,154 @@
+from .http_blocks.rest.rest_block import RESTPolling
+from nio.metadata.properties.string import StringProperty
+from nio.metadata.properties.timedelta import TimeDeltaProperty
+from nio.common.signal.base import Signal
+from datetime import datetime
+import requests
+
+
+class InstagramSearchByBase(RESTPolling):
+
+    """ This block polls the Instagram API, searching for all posts
+    by the specified users.
+
+    Params:
+        client_id (string): api credentials.
+        lookback (timedelta): amount of time to lookback for posts on start.
+
+    """
+
+    client_id = StringProperty(name="Client ID",
+                               default="[[INSTAGRAM_CLIENT_ID]]")
+    lookback = TimeDeltaProperty()
+
+    def __init__(self):
+        super().__init__()
+        self._created_field = 'created_time'
+
+    def configure(self, context):
+        super().configure(context)
+        lb = self._unix_time(datetime.utcnow() - self.lookback)
+        self._freshest = [lb] * self._n_queries
+        # Convert queries from usernames to ids.
+        self.queries = [i for i in [self._process_query(q)
+                        for q in self.queries] if i]
+        # reset n in case some usernames did not convert to ids.
+        self._n_queries = len(self.queries) or 1
```

Comment:
I think this "or 1" used to exist in the rest polling block and was copied over from that, but I don't think it really makes sense. If their are no queries then it should be 0.